lib/vsc/filesystem/quota/entities.py

```diff
@@ -0,0 +1,182 @@
+# it under the terms of the GNU Library General Public License as
+# published by the Free Software Foundation, either version 2 of
+# the License, or (at your option) any later version.
+#
+# vsc-filesystems-quota is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Library General Public License for more details.
+#
+# You should have received a copy of the GNU Library General Public License
+# along with vsc-filesystems-quota. If not, see <http://www.gnu.org/licenses/>.
+#
+"""Various classes that can represent quota information for the various
+devices an entity (user, vo, ...) uses on the VSC storage.
+
+@author: Andy Georges (Ghent University)
+"""
+from collections import namedtuple
+
+
+QuotaInformation = namedtuple('QuotaInformation', [
+    'timestamp',  # timestamp of the recording moment
+    'used',  # used quota in KiB
+    'soft',  # soft quota limit in KiB
+    'hard',  # hard quota limit in KiB
+    'doubt',  # the KiB GPFS is not sure about
+    'expired',  # tuple (boolean, grace period expressed in seconds)
+    'files_used',  # used number of inodes
+    'files_soft',  # soft limit for inodes
+    'files_hard',  # hard limit for inodes
+    'files_doubt',  # the inodes GPFS is not sure about
+    'files_expired',  # tuple (boolean, grace period expressed in seconds)
+])
+
+
+class QuotaEntity(object):
+    """Definition of a entity with associated quota.
+
+    The relevant information is stored in a RUDict instance,
+    with the following structure:
+    - string filesystem (name as known to GPFS)
+        - string fileset (name as known to GPFS)
+            - QuotaInformation
+    """
+    def __init__(self, storage, filesystem):
+        """ Initialiser """
+        self.storage = storage
+        self.filesystem = filesystem
+        self.quota_map = {}
+        self.exceed_map = {}
+        self.exceed = False
+
+    def update(self, fileset, used=0, soft=0, hard=0, doubt=0, expired=(False, None),
+               files_used=0, files_soft=0, files_hard=0, files_doubt=0, files_expired=(False, None),
+               timestamp=None):
+        """Store the quota for a given device.
+
+        The arguments to this function are turned into a recursive
+        dictionary that can easily be added to the existing information.
+        """
+
+        self.quota_map[fileset] = QuotaInformation(
+            timestamp=timestamp,
+            used=used,
+            soft=soft,
+            hard=hard,
+            doubt=doubt,
+            expired=expired,
+            files_used=files_used,
+            files_soft=files_soft,
+            files_hard=files_hard,
+            files_doubt=files_doubt,
+            files_expired=files_expired,
+        )
+
+        self.exceed = self.exceed or expired[0] or files_expired[0]
+
+    def exceeds(self):
```

Comment:
not sure why you have a method to access an anttribute? you expect to implement more logic?